### DISCUSSION TITLE
Taskwarrior Mutex

### DISCUSSION THREAD

**Discussion by tom-doerr** [GothenburgBitFactory/taskwarrior#2559] [Q&A] [answered]
Is it save to execute multiple taskwarrior commands in parallel or may I end up corrupting or overwriting data?

**Comment by tbabej** [comment, marked as answer]
Depends. Taskwarrior implements file locking mechanism using `fcntl`, its implementation can be found in libshared's `FS.cpp` (see `File::lock`). This works on most filesystems, though there are notable exceptions, like NFSv3.

Note for report commands you can use `rc.gc:0` to prevent writes to data files.

**Reply by tom-doerr** [reply]
Thanks! That explains why I used to have issues on ZFS and can't reproduce them now on Ext4.

**Reply by tbabej** [reply]
You're welcome! Support for atomic file writes (@vrusinov is spearheading this, follow #152 for changes) in Taskwarrior should make this even more robust.
